packages/lab/src/date-picker/DatePickerRangeGridPanel.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  FlexItem,
+  FlexLayout,
+  FormFieldContext,
+  type FormFieldContextValue,
+  FormFieldHelperText,
+  StackLayout,
+  makePrefixer,
+  resolveResponsiveValue,
+  useBreakpoint,
+  useControlled,
+} from "@salt-ds/core";
+import type { DateFrameworkType } from "@salt-ds/date-adapters";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import clsx from "clsx";
+import { type SyntheticEvent, forwardRef, useCallback, useState } from "react";
+import {
+  CalendarGrid,
+  CalendarNavigation,
+  type CalendarRangeProps,
+} from "../calendar";
+import { Calendar, type DateRangeSelection } from "../calendar";
+import { useLocalization } from "../localization-provider";
+import {
+  type RangeDatePickerState,
+  useDatePickerContext,
+} from "./DatePickerContext";
+import datePickerPanelCss from "./DatePickerPanel.css";
+import type { DatePickerPanelBaseProps } from "./DatePickerSingleGridPanel";
+
+const withBaseName = makePrefixer("saltDatePickerPanel");
+
+/**
+ * Props for the DatePickerRangeGridPanel component.
+ * @template TDate - The type of the date object.
+ */
+export type DatePickerRangeGridPanelProps<TDate extends DateFrameworkType> =
+  DatePickerPanelBaseProps<TDate> &
+    DateRangeSelection<TDate> & {
+      onSelectionChange?: (
+        event: SyntheticEvent,
+        selectedDate?: DateRangeSelection<TDate> | null,
```

Comment:
This reminded me the need to document undefined vs null

Reply:
will add to the list